C3 Custom Chart

vidma opened the pull request:
- [x] c3 chart. get autoresize for free (in report-view mode)
- [x] ability to pass optional `chartOptions`
- [x] some C3 charts are messed up (e.g. linechart)
  * looks like CSS conflict...

<img width="1121" alt="screen shot 2015-12-07 at 17 50 11" src="https://cloud.githubusercontent.com/assets/213426/11631529/143ecb02-9d0b-11e5-8ac6-2569d01fa403.png">

@andypetrella

vidma commented on notebooks/sql/DataFrame simple c3.snb:
```diff
@@ -0,0 +1,83 @@
+{
```
this file will likely go to `viz/Custom charts with C3.snb`

andypetrella commented on modules/common/src/main/scala/notebook/front/widgets/package.scala:
```diff
@@ -506,6 +506,16 @@ package object widgets {
                                                   "width" → sizes._1, "height" → sizes._2)))
   }
 
+  case class CustomC3Chart[C:ToPoints:Sampler](originalData:C, chartOptions :String = "{}", override val sizes:(Int, Int)=(600, 400), maxPoints:Int = 25) extends Chart[C] {
```
cool! however wouldn't be better to allow passing a `JsValue` that defaults to `Json.obj()`?

vidma replied:
not sure. 

- `JsValue` is Play-specific, so people would get scared ;) anyway they'd most likely [create it by parsing  regular json](https://www.playframework.com/documentation/2.2.x/ScalaJson#Using-string-parsing).
- maybe something similar could be useful for defining & combining a few preset `configurations`, like `stacked bar chart` with custom options like `column-name for x-axis`. But not convinced yet.

vidma commented:
styles for some chart types are messed up, e.g. line chart (no clue how to fix this yet):

<img width="814" alt="screen shot 2015-12-08 at 12 39 10" src="https://cloud.githubusercontent.com/assets/213426/11653734/e2e1b9cc-9da8-11e5-8b77-465b28f3c783.png">

vidma commented:
updated docs.
if not for issue with line charts, it is ready to :rocket:

andypetrella commented:
it was a css clash right? I saw that in the pivot (I fixed the bar at that time I think)

vidma commented:
> it was a css clash right? I saw that in the pivot (I fixed the bar at that time I think)

something like that, I just generalized and moved your fix into CSS, didn't go deep into details.
